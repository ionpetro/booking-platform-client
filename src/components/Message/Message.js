const types = {
  success: 'success',
  warning: 'warning',
  error: 'error'
};

export default {
  props: {
    type: {
      type: [String],
      validator: value => Object.values(types).includes(value)
    }
  }
};
