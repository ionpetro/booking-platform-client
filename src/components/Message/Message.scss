.message {
  display: flex;
  padding: 13px 14px;
  width: 480px;
  height: 48px;
  border-radius: 20px;
  margin-top: 30px;

  &.error {
    background-color: $red-tint;
    color: $red-dark;
  }
  &.warning {
    background-color: $yellow-tint;
    color: $yellow-dark;
  }
  &.success {
    background-color: $green-tint;
    color: $green-dark;
  }
  &__text {
    padding-left: 15px;
  }

  &__icon {
    // mobile
    @media #{$mobile} {
      margin: auto 0;
    }
  }

  // mobile
  @media #{$mobile} {
    width: 100%;
    font-size: 14px;
  }

  // vertical align on extra small screens
  @media only screen and (max-width: 450px) {
    height: auto;
  }
}
