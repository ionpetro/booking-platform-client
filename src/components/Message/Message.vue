<template>
  <div id="message" :class="type">
    <div id="icon">
      <img v-if="type === 'error'" src="@/assets/icons/error.svg" alt="error-icon" />
      <img v-else-if="type === 'warning'" src="@/assets/icons/warning.svg" alt="warning-icon" />
      <img v-else-if="type === 'success'" src="@/assets/icons/success.svg" alt="success-icon" />
    </div>
    <div id="text">
      <slot></slot>
    </div>
  </div>
</template>

<script src="./Message.js"></script>

<style src="./Message.scss" lang="scss"></style>
