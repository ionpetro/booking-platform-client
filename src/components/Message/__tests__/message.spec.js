import {mount} from '@vue/test-utils';
import Message from '../Message.vue';

describe('Message', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Message, {
      propsData: {
        type: 'error'
      }
    });
  });

  describe('when loaded', () => {
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
