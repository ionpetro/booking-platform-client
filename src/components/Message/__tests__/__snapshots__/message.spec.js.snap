// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message when loaded renders correctly 1`] = `
<div
  class="message error"
>
  <div
    class="message__icon"
  >
    <img
      alt="error-icon"
      src="@/assets/icons/error.svg"
    />
  </div>
   
  <div
    class="message__text"
  />
</div>
`;
